esp/esp/users/models/__init__.py

```diff
@@ -2030,14 +2030,14 @@ def create_from_Q(item_model, q_filter, description = ''):
         dumped_filter = pickle.dumps(q_filter)
 
         # Deal with multiple instances
-        query_q = Q(item_model = str(item_model), q_filter = dumped_filter, sha1_hash = hashlib.sha1(dumped_filter).hexdigest())
+        query_q = Q(item_model = str(item_model), q_filter = dumped_filter, sha1_hash = hashlib.sha1(dumped_filter.encode('UTF-8')).hexdigest())
```

Comment:
pickle.dumps produces a sequence of bytes, not a string, and should not be encoded with UTF-8.  Same with all the other instances where you hash dumped_filter (I've only commented on this one).